### PR TITLE
doc: some extensions and Doxygen cleanup

### PR DESCRIPTION
**Pull request by gmarull**
```
doc: extensions: link-roles: remove usage of local_util
local_util module is no longer used by other extensions, so remove its
usage.
```
```
doc: extensions: delete unused local_util
The local_util utilities are no longer used, so remove them.
```
```
doc: doxygen: prefix Doxygen utils dir with underscore
The same convention is used for Sphinx files.
```
```
doc: doxygen: delete unused custom footer template
The footer template is not used, so remove it.
```
```
doc: move favicon to static folder
Similar to other assets, e.g. logo, move favicon to the static folder.
```
```
doc: doxygen: move logo to Doxygen assets folder
Move the logo used by Doxygen into the Doxygen assets folder.
```
```
doc: remove unused images
Some Zephyr logo images were not used, so remove them.
```

### CODE REVIEW

**Comment by gmarull** [comment]
@nashif @carlescufi ping

File: doc/_scripts/redirects.py [hunk 1]
```diff
@@ -13,136 +13,6 @@
 """
 
 REDIRECTS = [
-    ('api/api', 'reference/index'),
```
Comment (mbolivar-nordic) [inline doc/_scripts/redirects.py:16]: I'm not sure about this one; seems nice to have these old redirects in place. Maybe separate them into an OLD_REDIRECTS variable that gets appended onto REDIRECTS for clarity for now instead?

**Review by mbolivar-nordic — COMMENTED** [review]
One qualm about the last patch, otherwise LGTM.

File: doc/_scripts/redirects.py (hunk 1 above)
Comment (gmarull) [reply doc/_scripts/redirects.py:16]: I'm not sure if it's worth to keep a list of redirects for docs we do not list (deleted ones apply to ≤ 1.13.0):

image

Comment (mbolivar-nordic) [reply doc/_scripts/redirects.py:16]: I understand they're not listed but people still have old links in their bookmarks.

Comment (gmarull) [reply doc/_scripts/redirects.py:16]: I think we should have a certain policy here, we should definitely avoid keeping old redirects forever. @nashif ?

Comment (mbolivar-nordic) [reply doc/_scripts/redirects.py:16]: We can always stick them in another file that we load from conf.py if you're worried about the size of the list. It seems like a cheap and easy thing to do with minimal maintenance burden, no?

Comment (gmarull) [reply doc/_scripts/redirects.py]: Dropped the commit for now, I prefer to not block other changes while this is decided.

**Review by cfriedt — APPROVED** [review]
LGTM. Is `footer.html` automatically generated?

**Comment by gmarull** [comment]
> LGTM. Is `footer.html` automatically generated?

I imagine it does if not provided, since I see some footer content here: https://docs.zephyrproject.org/apidoc/latest/
The file was likely there because Doxygen writes out all customizable files when invoked: https://www.doxygen.nl/manual/customize.html

**Review by mbolivar-nordic — APPROVED** [review]

**Pull request merged**